programs/bpf_loader/src/lib.rs

```diff
@@ -473,14 +473,16 @@ fn process_instruction_common(
         ic_logger_msg!(log_collector, "Program is not executable");
         return Err(InstructionError::IncorrectProgramId);
     }
+
     let programdata_account = if bpf_loader_upgradeable::check_id(program_account.get_owner()) {
-        let programdata_account = instruction_context.try_borrow_program_account(
-            transaction_context,
-            instruction_context
-                .get_number_of_program_accounts()
-                .saturating_sub(2),
-        )?;
-        Some(programdata_account)
+        instruction_context
+            .try_borrow_program_account(
+                transaction_context,
+                instruction_context
+                    .get_number_of_program_accounts()
+                    .saturating_sub(2),
+            )
+            .ok()
```

Comment:
As this is only an `Option` type this could lead to a confusion between the upgradeable loader vs the normal loader and whether a program was loaded in a transaction or not (e.g. used for an instruction or has the `is_writable` flag in the message).

Reply:
`load_program_from_account()` call preceding this code block is always going to return the program entry from the cache. This code block will be deleted in the follow up PR, and replaced with the cache lookup.

Would you prefer if the code deletion/refactor was done in this PR itself. It was causing some changes to tests in other parts of the code which are unrelated to this PR.